### PR TITLE
Add more unit-test test cases

### PR DESCRIPTION
**Pull request by scalarindetail**
#7 aims to add more and complete the unit-test cases.

### CODE REVIEW

File: test/signature_signer.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    });
+
+    it('throws exception when pem file is not correct', () => {
+      const signer = new JsrsasignSignatureSigner('incorrect pem');
+      expect(() => {
+        signer.sign('content');
+      }).to.throw('Failed to load private key');
+    });
+
+    it('throws exception when it not is ASN.1 hex string', () => {
```
Comment (Torch3333) [inline test/signature_signer.test.js]:
```suggestion
    it('throws exception when it is not an ASN.1 hex string', () => {
```

File: test/client_service_base.test.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        const request = await service.registerContract('contractId', 'foo',
+            new Uint8Array([1, 2, 3]), clientProperties);
+        assert(mockSpySetContractId.calledOnce);
+        assert(mockSpyContractRegistrationRequest.calledOnce);
+        assert(mockSpySetContractBinaryName.calledOnce);
+        assert(mockSpySetContractByteCode.calledOnce);
+        assert(mockSpySetContractProperties.calledOnce);
+        assert(mockSpySetCertHolderId.calledOnce);
+        assert(mockSpySetCertVersion.calledOnce);
+        assert(mockSpySetSignature.calledOnce);
```
Comment (Torch3333) [inline test/client_service_base.test.js]: So as to make the test more robust, would it be possible to assert that the spied on method was called with an expected parameter ?
For example, for the `mockSpySetContractId`, can we verify the method was called once with the parameter `'contractId'`.
If it is possible, let's do the same thing for other tests.

File: test/signature_signer.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('should work properly', () => {
+      const pem = fs.readFileSync(__dirname + '/key.pem', 'utf-8');
+      const signer = new JsrsasignSignatureSigner(pem);
+      assert.instanceOf(signer.sign('Content'), Uint8Array);
+    });
+  });
+
+  describe('Class EllipticSignatureSigner', () => {
+    it('throws exception when pem file is not correct', () => {
```
Comment (Torch3333) [inline test/signature_signer.test.js]:
```suggestion
    it('throws exception when the pem file is not correct', () => {
```

File: test/signature_signer.test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      const signer = new SignatureSigner('incorrect pem'); // EllipticSignature is used by default
+      expect(() => {
+        signer.sign('content');
+      }).
+          to.
+          throw(
+              'Failed to sign the request unsupported PKCS#1/5 hexadecimal key');
```
Comment (Torch3333) [inline test/signature_signer.test.js]: The indentation format seem different than above. Is it correct ?

Comment (scalarindetail) [reply test/signature_signer.test.js]: No, but i fixed it. Thanks!

File: test/client_service_base.test.js [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+        const mockSpyLedgerServiceResponse = sinon.spy(mockedProtobuf,
+            'LedgerServiceResponse');
+        const service = new ClientServiceBase(
+            ledgerClient, mockedProtobuf, clientProperties);
+        await service.sendRequest('registerCert', () => {
+          throw new Error();
+        });
+        sinon.assert.calledOnce(mockSpyLedgerServiceResponse);
+        sinon.assert.calledOnce(mockSpySetMessage);
+        sinon.assert.calledOnce(mockSpySetStatus);
```
Comment (Torch3333) [inline test/client_service_base.test.js]: Could you assert the values set here too ?

**Comment by Torch3333** [comment]
@feeblefakie 
There was further modifications after your last review, if you are ok with it the PR can be merged.

**Comment by feeblefakie** [comment]
@moon004  @Torch3333 Thank you ! It looks better with the additional changes. Merging...

**Pull request merged**